clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -2638,6 +2638,28 @@ def SinOp : UnaryFPToFPBuiltinOp<"sin">;
 def SqrtOp : UnaryFPToFPBuiltinOp<"sqrt">;
 def TruncOp : UnaryFPToFPBuiltinOp<"trunc">;
 
+//===----------------------------------------------------------------------===//
+// Expect op
+//===----------------------------------------------------------------------===//
+
+def ExpectOp : CIR_Op<"expect", [Pure, AllTypesMatch<["result", "val", "expected"]>]> {
+  let summary = "Provides information about expected value of val";
+  let description = [{
+    Same as llvm.expect and llvm.expect.with.probability intricnsics.
```

Comment:
This isn't about LLVM necessarily. I suggest you start with something like `Provides  __builtin_expect functionality ...` and then start a new paragraph to mention what the LLVM lowering expectations are.